client/src/components/NavHeader.js

```diff
@@ -1,25 +1,71 @@
-import React from "react";
-import { Container, Navbar, Nav } from "react-bootstrap";
+import React, { Component } from "react";
+import {
+  Button,
+  ButtonGroup,
+  Dropdown,
+  Container,
+  Navbar,
+  Nav,
+} from "react-bootstrap";
 import { LinkContainer } from "react-router-bootstrap";
 
-function NavHeader() {
-  return (
-    <Navbar bg="light">
-      <Container>
-        <LinkContainer to="/">
-          <Navbar.Brand>Blossym</Navbar.Brand>
-        </LinkContainer>
-        <Nav className="mr-auto">
-          <LinkContainer to="/fan">
-            <Nav.Link>Fan</Nav.Link>
-          </LinkContainer>
-          <LinkContainer to="/creator">
-            <Nav.Link>Creator</Nav.Link>
+class NavHeader extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      creatorAddress: "",
+      amountValue: 0,
+    };
+  }
+
+  render() {
+    let walletButton;
+    let formattedAddress;
+    if (!this.props.connectedWallet) {
+      walletButton = (
+        <Button variant="primary" onClick={this.props.onWalletConnectClick}>
+          Connect Wallet
+        </Button>
+      );
+    } else {
+      formattedAddress =
+        this.props.connectedWallet.substring(0, 5) +
+        "..." +
+        this.props.connectedWallet.substring(39);
```

Comment:
Just wondering, was this to keep the button shorter width-wise?

Optional: Increase left side of `...` to 4 characters after 0x and 4 characters after `...` like Uniswap's:
![Screen Shot 2021-01-26 at 11 29 02 PM](https://user-images.githubusercontent.com/2449384/105958085-ff204880-602e-11eb-85b5-c0670daa6a12.png)

More letters in the address can't hurt when users wanna double check they're using the right account 😆

Reply:
Let's discuss this later. I think we can do even more to tweak the UI... maybe add an icon too